pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -160,15 +165,28 @@ def __init__(self, mps, wires, id=None):
         assert qml.math.isclose(
             new_dj0, dj2
         ), "Dimension mismatch: the last tensor's first dimension does not match the previous third dimension."
-        super().__init__(*mps, wires=wires, id=id)
+
+        self.hyperparameters["input_wires"] = qml.wires.Wires(wires)
+
+        if work_wires:
+            self.hyperparameters["work_wires"] = qml.wires.Wires(work_wires)
+            all_wires = self.hyperparameters["input_wires"] + self.hyperparameters["work_wires"]
+        else:
+            self.hyperparameters["work_wires"] = None
+            all_wires = self.hyperparameters["input_wires"]
+
+        super().__init__(*mps, wires=all_wires, id=id)
 
     @property
     def mps(self):
         """list representing the MPS input"""
         return self.data
 
     def _flatten(self):
-        hyperparameters = (("wires", self.wires),)
+        hyperparameters = (
+            ("wires", self.hyperparameters["input_wires"]),
+            ("work_wires", self.hyperparameters["work_wires"]),
+        )
```

Comment:
Can we make this a dictionary so we don't have to do `hyperparams_dict = dict(metadata)`?

Reply:
It has to be done in this way to make sure the output form flatten is hashable